### PR TITLE
Server Side Encryption to Amazon S3

### PR DESCRIPTION
**Pull request by william-gr**
Ticket: #25621

### CODE REVIEW

File: src/middlewared/middlewared/plugins/backup.py [hunk 1]
```diff
@@ -87,17 +90,24 @@ class BackupService(CRUDService):
     async def query(self, filters=None, options=None):
         return await self.middleware.call('datastore.query', 'tasks.cloudsync', filters, options)
 
-    async def _clean_credential(self, data):
+    async def _clean_credential(self, verrors, name, data):
+
         credential = await self.middleware.call('datastore.query', 'system.cloudcredentials', [('id', '=', data['credential'])], {'get': True})
-        assert credential is not None
+        if credential is None:
+            verrors.add(f'{name}.credential', f'Credential {data["credential"]} not found', errno.EEXIST)
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/backup.py]: Again, I don't think this is a good idea to report "File already exists" in this case.

Comment (william-gr) [reply src/middlewared/middlewared/plugins/backup.py]: @themylogin right, changed to ENOENT

File: src/middlewared/middlewared/plugins/backup.py [hunk 2]
```diff
@@ -87,17 +90,24 @@ class BackupService(CRUDService):
     async def query(self, filters=None, options=None):
         return await self.middleware.call('datastore.query', 'tasks.cloudsync', filters, options)
 
-    async def _clean_credential(self, data):
+    async def _clean_credential(self, verrors, name, data):
+
         credential = await self.middleware.call('datastore.query', 'system.cloudcredentials', [('id', '=', data['credential'])], {'get': True})
-        assert credential is not None
+        if credential is None:
+            verrors.add(f'{name}.credential', f'Credential {data["credential"]} not found', errno.ENOENT)
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/backup.py]: "No such file or directory" does not suit us either. I've thought of leaving EINVAL here (and everywhere else) or adding custom errno (if you need to catch this exact error from client somewhere else)

Comment (william-gr) [reply src/middlewared/middlewared/plugins/backup.py]: @themylogin I dont agree with that. It is just an errno, it does not necessarily translate to the string "No such file or directory". We can't add a errno for every single string/purpose. We define ourselves what they mean, semantically.

Comment (william-gr) [reply src/middlewared/middlewared/plugins/backup.py]: @themylogin can we agree to disagree and approve the PR? We can revisit the errno issue at a later date ;)

**Pull request merged**